src/udp.c

```diff
@@ -233,11 +233,48 @@ udp_bind ( int subsystem, const char *name,
       }
 #endif
 
-      if (setsockopt(fd, udp_get_solip(), IP_ADD_MEMBERSHIP, &m, sizeof(m))) {
-        inet_ntop(AF_INET, &m.imr_multiaddr, buf, sizeof(buf));
-        tvhwarn(subsystem, "%s - cannot join %s [%s]",
-                name, buf, strerror(errno));
+#if defined(PLATFORM_LINUX)
```

Comment:
Done, since this means that udp_get_ifaddr() is then potentially used on platforms other than OSX and Linux, I checked for the SIOCGIFADDR as well - it seems to exists universally, so I guess that's OK.

If you prefer a new PR with a single commit let me know.